### DISCUSSION TITLE
Arduino Mega2560 - Delay/Lag problem on analog inputs

### DISCUSSION THREAD

**Discussion by PaulHubiss** [shanteacontrols/OpenDeck#124] [Q&A]
Hi Paradajz :)
First of all, I would like to thank you very much for the interesting OpenDeck project. I've been looking for a DIY project to build a MIDI controller for a long time, and I really like your concept. I started testing, but I ran into a strange problem.

The reactions of the potentiometers on the analog inputs have a long delay. I recorded this behavior on video: https://youtu.be/Ja8fG_sWlso

I also tried MIDI mapping in Ableton and there is also a big delay :(

Can't you have a solution, please?

I apologize for my English, I'm from the Czech Republic :)

**Comment by paradajz** [comment]
Ouch, that seems bad. Need to check which change caused it. I'm in the middle of larger optimization cycle anyways geared towards AVR boards in general so there will be a new release soon.

Which firmware have you flashed, by the way?

**Reply by PaulHubiss** [reply]
Great, that sounds good ;)
I have uploaded the latest version of FW - 6.2.4
But I registered the same behavior with the previous version of FW.

**Reply by paradajz** [reply]
I will need to redesign I2C for this to work as expected. Might take a while. In the meantime, the "fix" is to disable the screen.

**Reply by PaulHubiss** [reply]
Great, thanks, this solution works and for now is enough for me. I wish you good luck in your further work on the project ;)

**Comment by paradajz** [comment]
This should now be solved - the changes are on master branch, but not released yet.

**Comment by paradajz** [comment]
Released: https://github.com/shanteacontrols/OpenDeck/releases/tag/v6.3.1

**Reply by PaulHubiss** [reply]
I apologize for the late reply, I confirm the functionality ;) Thanx!
